### PR TITLE
Implement offerRescinded callback

### PR DESCRIPTION
**Pull request by icexelloss**

### CODE REVIEW

File: scheduler/src/cook/mesos/scheduler.clj [hunk 1]
```diff
@@ -925,9 +934,12 @@
         (log/debug "Preparing to pull offer (offer-matcher)")
         (let [offer (async/<!! matcher-chan)
               fid (:fid offer)
+              offer-ids (:ids offer)
               driver (:driver offer)]
           (log/debug "(offer-matcher) Offer taken " offer)
-          (handle-resource-offer! conn driver fid pending-jobs-atom offer))
+          (if-let [invalid-offer-id (some #(cache/lookup @rescinded-offer-id-cache %) offer-ids)]
+            (log/info "Received rescinded offer" invalid-offer-id "in offer matcher")
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/scheduler.clj]: Add a meter metric for this so we can track the rate that this happens

File: scheduler/src/cook/mesos/scheduler.clj [hunk 2]
```diff
@@ -101,6 +102,14 @@
              "UTF-8")
      executor-key executor-value}))
 
+(defn rescind-offer!
+  [rescinded-offer-id-cache offer-id]
+  (log/info "Rescinding offer" offer-id)
+  (swap! rescinded-offer-id-cache (fn [c]
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/scheduler.clj]: Add a meter here so we know how often offers are rescinded

File: scheduler/src/cook/mesos/scheduler.clj (hunk 1 above)
Comment (icexelloss) [inline scheduler/src/cook/mesos/scheduler.clj]: Isn't this the same as the rescinding rate?

Comment (wyegelwel) [inline scheduler/src/cook/mesos/scheduler.clj]: I'm not sure =)

Comment (wyegelwel) [inline scheduler/src/cook/mesos/scheduler.clj]: I'm not sure. At the very least, the diff in time will give us information.

**Comment by icexelloss** [comment]
Due to resource constraints, I am deferring metrics to a later mr

**Pull request merged**